coreclient/src/clients/mod.rs

```diff
@@ -691,6 +691,52 @@ impl CoreUser {
         notifier.notify();
         Ok(value)
     }
+
+    /// This function goes through all tables of the database and returns all columns that contain the query.
+    pub async fn scan_database(&self, query: &str, strict: bool) -> anyhow::Result<Vec<String>> {
```

Comment:
Would it be possible to move this function to tests? It is not used in the app, right?